sources/core/Stride.Core/Serialization/Binary/BinarySerializationReader.cs

```diff
@@ -33,57 +34,85 @@ public override void Serialize(ref bool value)
             value = result != 0;
         }
 
-#pragma warning disable CS0618 // Type or member is obsolete
         /// <inheritdoc />
         public override unsafe void Serialize(ref float value)
         {
-            fixed (float* valuePtr = &value)
-                *((uint*)valuePtr) = UnderlyingStream.ReadUInt32();
+            Span<byte> buffer = MemoryMarshal.Cast<float, byte>(MemoryMarshal.CreateSpan(ref value, 1));
+            var read = UnderlyingStream.Read(buffer);
+            if (read != sizeof(float))
+                throw new EndOfStreamException();
+            value = BinaryPrimitives.ReadSingleLittleEndian(buffer);
```

Comment:
Is this still necessary? The read should directly populate the memory of the `value` - applies to every method.